server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ChunkOutputStream.java

```diff
@@ -33,39 +32,42 @@
 
    private final ReactorNettyHttpResponse parentResponse;
 
-    /**
-     * This is the {@link Mono} that we return from
-     * {@link ReactorNettyJaxrsServer.Handler#handle(HttpServerRequest, HttpServerResponse)}
-     */
-   private final MonoProcessor<Void> completionMono;
+   /**
+    * This is the {@link Mono} that we return from
+    * {@link ReactorNettyJaxrsServer.Handler#handle(HttpServerRequest, HttpServerResponse)}
+    */
+   private final Sinks.Empty<Void> completionSink;
 
-    /**
-     * Indicates that we've starting sending the response bytes.
-     */
-    private volatile boolean started;
+   /**
+    * Indicates that we've starting sending the response bytes.
+    */
+   private volatile boolean started;
 
-    /**
-     * This is ultimately think 'sink' that we write bytes to in {@link #asyncWrite(byte[], int, int)}.
-     */
-    private FluxSink<Tuple2<byte[], CompletableFuture<Void>>> byteSink;
+   /**
+    * This is ultimately think 'sink' that we write bytes to in
+    * {@link #asyncWrite(byte[], int, int)}.
+    */
+   private Sinks.Many<Tuple2<byte[], CompletableFuture<Void>>> byteSink;
 
-    /**
-     * This is used to establish {@link #byteSink} upon the first writing of bytes.
-     */
-   private final Supplier<FluxSink<Tuple2<byte[], CompletableFuture<Void>>>> byteSinkSupplier;
+   /**
+    * This is used to establish {@link #byteSink} upon the first writing of bytes.
+    */
+   private final Supplier<Sinks.Many<Tuple2<byte[], CompletableFuture<Void>>>> byteSinkSupplier;
+
+   private final EmitFailureHandler emitFailureHandler = EmitFailureHandler.FAIL_FAST;
```

Comment:
I just noticed that this is member variable.  Did you think about making it a static variable?

Reply:
Why would you prefer it to be a static variable in this case?

Reply:
To only maintain one pointer in memory as opposed to one per request/response.  Micro-optimization and probably eliminated by the runtime, but I see constants used in much more dubious places:)